{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Test](./images/general/logos_combined.png)\n",
    "\n",
    "# Maschine learning methods for predicting cell-cycle phase from scRNA-Seq data\n",
    "\n",
    "## By Ron Fechtner \n",
    "### Ludwig-Maximilians-Universit채t M체nchen / Technische Universit채t M체nchen\n",
    "\n",
    "\n",
    "This is the supplementary for the Bachelor's Thesis on Maschine learning methods for predicting cell-cycle phase from scRNA-Seq data.\n",
    "\n",
    "Here you will find the a digital copy of the thesis, the codes, data, images, restul tables and juypter notebooks. Most of the chapters have a associated jupyter notebook file within the notebook folder that shows how the results were generated. This Readme is an example of such an notebook. All code can be directly executed or modified. For further information please see: http://jupyter.readthedocs.io/en/latest/index.html\n",
    "\n",
    "If you don't want to install juypter notebook on your local machine you can visit http://nbviewer.jupyter.org/github/rfechtner/pypairs/tree/master/supplementary/notebooks/ for a online view (read only).\n",
    "\n",
    "The general structure of this folder is:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "/\n",
      "    Readme.ipynb\n",
      "    Readme.md\n",
      "\n",
      "\n",
      "code/\n",
      "    helper.py\n",
      "    pypairs.py\n",
      "    __init__.py\n",
      "\n",
      "    __pycache__/\n",
      "        helper.cpython-36.pyc\n",
      "        pypairs.cpython-36.pyc\n",
      "\n",
      "data/\n",
      "    biomart_human-genes.txt\n",
      "    biomart_mouse-human-orthologs.txt\n",
      "    cell_cycle_genes.csv\n",
      "    cyclebase_top1000_genes.tsv\n",
      "    E-MTAB-3929.processed.1_counts.txt\n",
      "    E-MTAB-3929_annotation.txt\n",
      "    E-MTAB-6142_human.csv\n",
      "    go_0007049_homoSapiens.csv\n",
      "    GSE53481_humanRNAseq.txt\n",
      "    GSE64016_H1andFUCCI_normalized_EC_human.csv\n",
      "    GSE71456_Samples_RPKM.csv\n",
      "    GSE75748_bulk_cell_type_ec.csv\n",
      "    GSE75748_bulk_time_course_ec.csv\n",
      "    GSE75748_sc_cell_type_ec.csv\n",
      "    GSE75748_sc_time_course_ec.csv\n",
      "    mESC_dataset_mouse.txt\n",
      "    mouse_pretrained-pairs.json\n",
      "    Non_norm.PolyA_NamedByAlex_human.csv\n",
      "    regev_lab_cell_cycle_genes.txt\n",
      "\n",
      "images/\n",
      "\n",
      "    application/\n",
      "        cell_lineage_e5-e7.pdf\n",
      "        ebv_circle.pdf\n",
      "        ebv_circle.png\n",
      "        ebv_line.pdf\n",
      "        ebv_line.png\n",
      "        ebv_scatter.pdf\n",
      "        prediction_e3-e7.pdf\n",
      "        prediction_e3-e7.png\n",
      "        prediction_e3-ee5.pdf\n",
      "        prediction_e3-ee5.png\n",
      "        prediction_e5-e7.pdf\n",
      "        prediction_e5-e7.png\n",
      "\n",
      "    differences/\n",
      "        sandbag-speed.pdf\n",
      "        sandbag-speed.png\n",
      "\n",
      "    evaluation/\n",
      "        E-MTAB-3929.sdrf.txt\n",
      "        hESC-scatter.pdf\n",
      "        hESC-scatter.png\n",
      "        hESC-scatter.svg\n",
      "        hPSC-assign-all-scatter.pdf\n",
      "        hPSC-assign-all.pdf\n",
      "        hPSC-assign-all.png\n",
      "        hPSC-assign-h1-scatter.pdf\n",
      "        hPSC-assign-h1.pdf\n",
      "        hPSC-assign-h1.png\n",
      "        ml_eval.pdf\n",
      "        ml_pred.pdf\n",
      "        mouse_on_human.pdf\n",
      "        mouse_on_human.png\n",
      "        mouse_on_human_norm.pdf\n",
      "        mouse_on_human_norm.png\n",
      "        oscope-fraction-test-cc-only.pdf\n",
      "        oscope-fraction-test-cc-only.png\n",
      "        oscope-fraction-test.pdf\n",
      "        oscope-fraction-test.png\n",
      "        oscope-fraction-test_old1.png\n",
      "        oscope-fraction-test_old2.png\n",
      "        oscope-pca.pdf\n",
      "        oscope-pca.png\n",
      "        oscope-phase-distribution.pdf\n",
      "        oscope-phase-distribution.png\n",
      "        prediction-mESC-on-hESC.png\n",
      "\n",
      "    extension/\n",
      "        networkx.pdf\n",
      "        networkx.png\n",
      "        pairs_ematb6142.pdf\n",
      "        pairs_ematb6142.png\n",
      "        rfp_ematb6142.pdf\n",
      "        rfp_ematb6142.png\n",
      "        rf_bulk.pdf\n",
      "        rf_bulk.png\n",
      "        triplets.pdf\n",
      "\n",
      "    general/\n",
      "        logo1.jpg\n",
      "        logo2.jpg\n",
      "        logos_combined.PNG\n",
      "\n",
      "notebooks/\n",
      "\n",
      "    2. Python reimplementation/\n",
      "        2.3 Differences in code - Python.ipynb\n",
      "        2.3 Differences in code - R.ipynb\n",
      "\n",
      "    3. Evaluation/\n",
      "        3.2 Mouse pairs on human dataset.ipynb\n",
      "        3.3 Internal cross validation.ipynb\n",
      "        3.4.1.1 Bulk - GSE53481.ipynb\n",
      "        3.4.1.2 Bulk - GSE71456.ipynb\n",
      "        3.4.2 Single cell - EMATB6142.ipynb\n",
      "\n",
      "    4. Application/\n",
      "        4.1 EBV.ipynb\n",
      "        4.2 E-MTAB-3929.ipynb\n",
      "\n",
      "    5. Extension/\n",
      "        5.1 Random forest on pairs.ipynb\n",
      "        5.2 Pairs network.ipynb\n",
      "        5.2.1 Weighted Pairs.ipynb\n",
      "        5.2.2. Weighted Triplets.ipynb\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "for root, dirs, files in os.walk(\"./\"):\n",
    "    level = root.replace(\"./\", '').count(os.sep)\n",
    "    indent = ' ' * 4 * (level)\n",
    "    print()\n",
    "    folder = os.path.basename(root)\n",
    "    if not folder.startswith(\".\"):\n",
    "        print('{}{}/'.format(indent, folder))\n",
    "        subindent = ' ' * 4 * (level + 1)\n",
    "        for f in files:\n",
    "            print('{}{}'.format(subindent, f))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### The PyPairs implementation is also available via pip:\n",
    "```shell\n",
    "$ pip install pypairs\n",
    "``` \n",
    "\n",
    "### And Github:\n",
    "https://github.com/rfechtner/pypairs\n",
    "\n",
    "### For questions please write me at:\n",
    "ronfechtner@gmail.com"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
